simulator/src/main/java/kcl/teamIndexZero/traffic/simulator/data/descriptors/TrafficLightsInSetRule.java

```diff
@@ -1,7 +1,24 @@
 package kcl.teamIndexZero.traffic.simulator.data.descriptors;
 
+import kcl.teamIndexZero.traffic.simulator.data.trafficLight.TrafficLight;
+import kcl.teamIndexZero.traffic.simulator.data.trafficLight.TrafficLightInSet;
+import kcl.teamIndexZero.traffic.simulator.data.trafficLight.TrafficLightState;
+
+import java.util.List;
+
 /**
  * Traffic Light in synchronous set rule
  */
 public class TrafficLightsInSetRule {
+
+    /**
+     * Changes colours at trafficLights in one set
+     *
+     * @param trafficLightList List of the traffic lights in one set
+     */
+    public static void changeStateofSet(List<TrafficLightInSet> trafficLightList){
```

Comment:
static? this means that this function only exists once no matter how many `TrafficLightsInSetRule` you have. Light change mechanics/logic should be in `TrafficController`. Rules just need to describe the what the logic must apply, not the logic itself. 
e.g.:  - Rule state that there are 2 lights that work synchronously in opposite state (when one is green the other is red). Rule also contains the ID of those two lights and the delay between changes.
- Controller takes this rule and creates 2 lights that start with opposite colour state.
- Controller's tick looks at the set and the delay inside of it then if the states needs changing then it goes through each of the lights and changes their state appropriately.